shared/src/business/useCaseHelper/caseAssociation/createUserForContactPrimary.test.js

```diff
@@ -0,0 +1,141 @@
+const {
+  applicationContext,
+} = require('../../test/createTestApplicationContext');
+const {
+  createUserForContactPrimary,
+} = require('./createUserForContactPrimary');
+const {
+  ROLES,
+  SERVICE_INDICATOR_TYPES,
+} = require('../../entities/EntityConstants');
+const { Case } = require('../../entities/cases/Case');
+const { MOCK_CASE } = require('../../../test/mockCase');
+
+describe('createUserForContactPrimary', () => {
+  const USER_ID = '674fdded-1d17-4081-b9fa-950abc677cee';
+
+  beforeEach(() => {
+    applicationContext.getUniqueId.mockReturnValue(USER_ID);
+  });
+
+  it('should throw an unauthorized error for non admissionsclerk users', async () => {
+    applicationContext.getCurrentUser.mockReturnValue({});
+
+    await expect(
+      createUserForContactPrimary({
+        applicationContext,
+        caseEntity: new Case(MOCK_CASE, { applicationContext }),
+        email: 'testing@example.com',
+        name: 'Bob Ross',
+      }),
+    ).rejects.toThrow('Unauthorized');
+  });
+
+  it('should call createNewPetitionerUser with the new user entity', async () => {
+    const UPDATED_EMAIL = 'testing@example.com';
+
+    applicationContext.getCurrentUser.mockReturnValue({
+      role: ROLES.admissionsClerk,
+    });
```

Comment:
Another one where all of these should probably go into `beforeEach(...`

Reply:
One of the tests checks for an empty currentUser.